src/common/some_type.h

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include <utility>
+
+namespace marian {
+
+class some_type {
```

Comment:
Is this a reimplementation of `boost::any`? How about `any_type`?

Reply:
Name changed